onnxruntime/core/providers/cpu/sequence/sequence_ops.cc

```diff
@@ -316,4 +319,229 @@ Status SequenceConstruct::Compute(OpKernelContext* context) const {
+                                          DataTypeImpl::GetTensorType<int32_t>(),
+                                          DataTypeImpl::GetTensorType<std::string>()})
+        .TypeConstraint("S", DataTypeImpl::AllSequenceTensorTypes())
+        .TypeConstraint("I", std::vector<MLDataType>{
+                                 DataTypeImpl::GetTensorType<int32_t>(),
+                                 DataTypeImpl::GetTensorType<int64_t>()}),
+    SplitToSequence);
+
+SplitToSequence::SplitToSequence(const OpKernelInfo& info) : OpKernel(info) {
+  axis_ = info.GetAttrOrDefault<int64_t>("axis", 0);
+  keepdims_ = info.GetAttrOrDefault<int64_t>("keepdims", 1);
+}
+
+Status SplitToSequence::Compute(OpKernelContext* context) const {
+  const Tensor& input = *context->Input<Tensor>(0);
+  const Tensor* p_split_input = context->Input<Tensor>(1);
+
+  Status status;
+  auto data_type = input.DataType();
+
+  if (data_type == DataTypeImpl::GetType<float>())
+    status = ComputeImpl<float>(*context, input, p_split_input);
+  else if (data_type == DataTypeImpl::GetType<int32_t>())
+    status = ComputeImpl<int32_t>(*context, input, p_split_input);
+  else if (data_type == DataTypeImpl::GetType<std::string>())
+    status = ComputeImpl<std::string>(*context, input, p_split_input);
+  else
+    ORT_THROW("SplitToSequence operator does not support ", data_type, " yet");
+
+  return status;
+}
+
+Status SplitToSequence::PrepareForCompute(const TensorShape& input_shape, int64_t split_scalar, bool is_split_input_scalar,
+                                          int64_t& num_outputs, int64_t& axis, int& before_dims,
+                                          int& after_dims_including_split_axis, int& after_dims_excluding_split,
+                                          bool& is_uneven_split, int& num_remaining_splits,
+                                          std::vector<int64_t>& split_sizes) const {
+  auto& input_dims = input_shape.GetDims();
+  const auto num_dimensions = gsl::narrow_cast<int64_t>(input_shape.NumDimensions());
+  axis = HandleNegativeAxis(axis_, num_dimensions);  // handle negative and enforce axis is valid
+  const int64_t split_dim_size = input_dims[axis];
+
+  before_dims = gsl::narrow<int>(input_shape.SizeToDimension(axis));
+  after_dims_including_split_axis = gsl::narrow<int>(input_shape.SizeFromDimension(axis));
+  after_dims_excluding_split = (axis + 1 == num_dimensions)
+                                   ? 1  // we multiply by this value so must be 1 not 0
+                                   : gsl::narrow<int>(input_shape.SizeFromDimension(axis + 1));
+
+  if (is_split_input_scalar) {
+    auto num_even_splits = split_dim_size / split_scalar;
+    num_remaining_splits = gsl::narrow_cast<int>(split_dim_size % split_scalar);
+    if (num_remaining_splits != 0) {
+      is_uneven_split = true;
+    }
+    num_outputs = num_even_splits + num_remaining_splits;
```

Comment:
`+ num_remaining_splits`

this should be + 1 if is_uneven_split.

This seems slipping through the test cases ...

Reply:
If split_dim_size = 17 and split_scalar = 3, then you'll have 2 remaining tensors. Hence I used ```+ num_remaining_splits```. The test "SplitToSequence_DefaultAxis0UnevenSplitFloat" tests the uneven split.

Reply:
> If 'split' is a scalar, then 'input' will be split into equally sized chunks(if possible). Last chunk will be smaller if the 'input' size along the given axis 'axis' is not divisible by 'split'.

That would imply splitting into [3, 3, 3, 3, 3, 2], where `num_outputs` is 5 + 1 = 6. This is inline with the code later that handles the split size of the last chunk https://github.com/microsoft/onnxruntime/blob/f27855648c9b259aebd35622f4a2df4073a0653e/onnxruntime/core/providers/cpu/sequence/sequence_ops.cc#L505
The current code results in `num_outputs` 5 + 2 = 7.

The test does not cover this case, because the last tensor chunk has split size 1.

Reply:
Thanks for catching this. Fixed in latest update.